dev/lint-python

```diff
@@ -26,30 +32,45 @@ PYLINT_REPORT_PATH="$SPARK_ROOT_DIR/dev/pylint-report.txt"
 PYLINT_INSTALL_INFO="$SPARK_ROOT_DIR/dev/pylint-info.txt"
 SPHINXBUILD=${SPHINXBUILD:=sphinx-build}
 SPHINX_REPORT_PATH="$SPARK_ROOT_DIR/dev/sphinx-report.txt"
+VIRTUALENV_EXEC="virtualenv"
+
+# It is a good practice to upgrade pip to the latest version, however, sometimes it might break
+# something. Let freeze this version here. We cannot put it into requirements.txt since the
+# execution of this requirement files might depends on the pip version.
+PIP_VERSION="8.1.2"
 
 cd "$SPARK_ROOT_DIR"
 
+DEBUG=false
+if [[ $1 == "--debug" || $1 == "-d" ]]; then
+    DEBUG=true
+fi
+
+
+if [[ ! -z $VIRTUAL_ENV  || ! -z $CONDA_PREFIX ]]; then
+    echo "We already are inside a virtual environment: $VIRTUAL_ENV$CONDA_PREFIX. Using it"
+else
+  if hash conda 2>/dev/null; then
+    echo "Not inside a virtual environment. Creating conda virtual env"
+    conda create -y -n $VIRTUAL_ENV_NAME python
```

Comment:
The problem is you create the conda environment inside the anaconda env dir (`/opt/anconda3/envs/`) and not in `./build/venv`. So you might have collision with existing environments, users might not like this,...

in anaconda, all env are stored at the same location